src/openssh/patch/0004-ssh-crash-fix.patch

```diff
@@ -0,0 +1,107 @@
+From 745b435a3bf705987ed546e32bd9fe0c6e5378f3 Mon Sep 17 00:00:00 2001
+From: Asha Behera <asha.behera@broadcom.com>
+Date: Mon, 24 Feb 2025 23:19:10 -0700
+Subject: [PATCH] [PATCH] ssh crash fix
+
+---
+ auth.c    | 9 +++++++--
+ auth.h    | 2 +-
+ packet.c  | 2 +-
+ session.c | 5 ++++-
+ sshd.c    | 6 ++++--
+ 5 files changed, 17 insertions(+), 7 deletions(-)
+
+diff --git a/auth.c b/auth.c
+index 73a85fd..4b4a7b2 100644
+--- a/auth.c
++++ b/auth.c
+@@ -1191,12 +1191,17 @@ auth_authorise_keyopts(struct ssh *ssh, struct passwd *pw,
+ }
+ 
+ /* Export remote IP address and port for authorization. */
+-void
++int
+ export_remote_info(struct ssh *ssh)
+ {
++        if (ssh == NULL)
++        {
++          return 1;
++        }
+ 	const char *remote_ip = ssh_remote_ipaddr(ssh);
+ 	const int remote_port = ssh_remote_port(ssh);
+ 	const char remote_addr_port[32 + INET6_ADDRSTRLEN];
+ 	snprintf(remote_addr_port, sizeof(remote_addr_port), "%s %d", remote_ip, remote_port);
+ 	setenv("SSH_CLIENT_IPADDR_PORT", remote_addr_port, 1);
+-}
+\ No newline at end of file
++        return 0; 
++}
+diff --git a/auth.h b/auth.h
+index a83d31d..7186177 100644
+--- a/auth.h
++++ b/auth.h
+@@ -234,7 +234,7 @@ pid_t	subprocess(const char *, struct passwd *,
+ int	 sys_auth_passwd(struct ssh *, const char *);
+ 
+ /* Export remote IP address and port for authorization. */
+-void	 export_remote_info(struct ssh *);
++int	 export_remote_info(struct ssh *);
+ 
+ #if defined(KRB5) && !defined(HEIMDAL)
+ krb5_error_code ssh_krb5_cc_gen(krb5_context, krb5_ccache *);
+diff --git a/packet.c b/packet.c
+index 00e3180..f53ce95 100644
+--- a/packet.c
++++ b/packet.c
+@@ -529,7 +529,7 @@ ssh_remote_ipaddr(struct ssh *ssh)
+ 	int sock;
+ 
+ 	/* Check whether we have cached the ipaddr. */
+-	if (ssh->remote_ipaddr == NULL) {
++	if (ssh != NULL && ssh->remote_ipaddr == NULL) {
+ 		if (ssh_packet_connection_is_on_socket(ssh)) {
+ 			sock = ssh->state->connection_in;
+ 			ssh->remote_ipaddr = get_peer_ipaddr(sock);
+diff --git a/session.c b/session.c
+index d0c7f36..973b58f 100644
+--- a/session.c
++++ b/session.c
+@@ -625,7 +625,10 @@ do_exec_pty(struct ssh *ssh, Session *s, const char *command)
+ 		close(ttyfd);
+ 
+ 		/* Export remote IP address and port for authorization. */
+-		export_remote_info(ssh);
++                if (export_remote_info(ssh) != 0)
++                {
++		  return -1;
++                }
+ 
+ 		/* record login, etc. similar to login(1) */
+ #ifndef HAVE_OSF_SIA
+diff --git a/sshd.c b/sshd.c
+index a2761f1..e14290a 100644
+--- a/sshd.c
++++ b/sshd.c
+@@ -1658,7 +1658,6 @@ main(int ac, char **av)
+ 			break;
+ 		case 'C':
+ 			/* Export remote IP address and port for authorization. */
+-			export_remote_info(ssh);
```

Comment:
@liuh-80 : The call of export_remote_info(ssh) is not right as ssh is assigned after call to ssh_packet_set_connection at line 2169 in file sshd.c .
I want to remove this API call to export_remote_info as it is a NOOP as ssh pointer will always be NULL here

Reply:
We need export_remote_info to export remote address for TACACS feature.
I will investigation and fix the issue, but we need keep the export_remote_info code.

Reply:
Please make sure all TACACS related test case does not break.
We need export_remote_info for TACACS per-command authorization.
I think you can keeps those export_remote_info invoke code no change, because the fix in export_remote_info already return when ssh is empty.

Reply:
But if it will always ssh argument to export_remote_info   will always be NULL, then the call to export_remote_info  is of no use. What is the usecase to keep the call here ?

Reply:
For remove export_remote_info at line 89, if the ssh always be NULL, I agree with you, this can be removed.

Reply:
keeping in intact, to keep changes minimal

Reply:
#Closed